packages/inventory/src/Stocktake/DetailView/DetailView.tsx

```diff
@@ -24,20 +23,14 @@ import { AppRoute } from '@openmsupply-client/config';
 import { StocktakeLineFragment } from '../api';
 
 export const DetailView: FC = () => {
-  const { isOpen, entity, onOpen, onClose, mode } = useEditModal<Item>();
+  const { isOpen, entity, onOpen, onClose, mode } =
+    useEditModal<ItemRowFragment>();
   const { data, isLoading } = useStocktake();
   const navigate = useNavigate();
   const t = useTranslation('inventory');
 
   const onRowClick = (item: StocktakeLineFragment | StocktakeSummaryItem) => {
-    if ('lines' in item) return onOpen(toItem(item));
-    onOpen(
-      toItem({
-        itemId: item.item?.id ?? '',
-        itemName: item.item?.name ?? '',
-        itemCode: item.item?.code ?? '',
-      })
-    );
+    if (item.item) onOpen(item.item);
```

Comment:
What I have changed is that a `StockLineFragment` has an `item` field with `{ id, code, name }`, which is equal to an `ItemRowFragment`. Then, the `SummaryItem` also has a `item` field with the same shape, so instead of trying to 'fix' the `toItem` function which the original issue was for, I just removed the need to use it